File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.cpp
class Solution{
    public:
        //中序：左 中 右
        vector<int> inorderTraversal(TreeNode* root){
            vector<int> result;
            inorder(root, result);
            return result;
        }

        void inorder(TreeNode* current, vector<int>& path){
            if(current == nullptr){
                return;
            }
            inorder(current->left, path);
            path.push_back(current->val);
            inorder(current->right, path);
        }
};

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        inOrder = []

        def traversal(node):
            if node is None:
                return

            traversal(node.left)
            inOrder.append(node.val)
            traversal(node.right)
        
        traversal(root)

        return inOrder